File: my-leetcode-solutions/valid-palindrome.ts
// My solution to 125 - Valid Palindrome

function isPalindrome(s: string): boolean {
    
};

const testCases = [
    "A man, a plan, a canal: Panama",
    "race a car",
    " "
];

for(let i = 0; i < testCases.length; i++) {
    const output = isPalindrome(testCases[i]);
    console.log(`Output: ${output} for input: ${testCases[i]}`);
}


export {};

File: my-leetcode-solutions-py/valid-palindrome.py
# My solution to 125 - Valid Palindrome

class Solution:
    def isPalindrome(self, s: str) -> bool:
        """
        convert to lower case
        remove all non-alphanumeric characters
        reads same forward and backward (use pointers)
        """

        lowercase = s.lower()

        alphanumArr = []
        for char in lowercase:
            if char.isalnum():
                alphanumArr.append(char)

        alphanumStr = "".join(alphanumArr)

        startPointer = 0
        endPointer = len(alphanumStr) - 1

        while startPointer < endPointer:
            if alphanumStr[startPointer] != alphanumStr[endPointer]:
                return False

            startPointer+=1
            endPointer-=1

        return True

        


# Time Complexity:

s_test_cases = [
    "A man, a plan, a canal: Panama",
    "race a car",
    " ",
]

solution = Solution()

for i in range(len(s_test_cases)):
    output = solution.isPalindrome(s_test_cases[i])
    print(f"Output: {output!r} for s as {s_test_cases[i]!r}")
